lib/resources/ms_message_queue.rb

```diff
@@ -0,0 +1,222 @@
+  class MicrosoftMessageQueue < Inspec.resource(1)
+    name 'ms_message_queue'
+    supports platform: 'windows'
+    desc <<-EOH
+      Provides visibility into a Microsoft Message Queue
+    EOH
+
+    example <<-EOH
+      describe ms_message_queue('.\\private$\\MyQueueHere') do
+        it { should exist }
+        its('QueueName') { should eq 'MyQueueHere' }
+        it { should be_allowed('full-control', by_user: 'MyComputerName\\MyAwesomeUser') }
+      end
+    EOH
+
+    PERMISSIONS = {
+      'full-control' => 'MQSEC_QUEUE_GENERIC_ALL',
+      'read' => 'MQSEC_QUEUE_GENERIC_READ',
+      'write' => 'MQSEC_QUEUE_GENERIC_WRITE',
+
+      'take-ownership' => 'MQSEC_TAKE_QUEUE_OWNERSHIP',
+      'delete-queue' => 'MQSEC_DELETE_QUEUE',
+      'delete-journal' => 'MQSEC_DELETE_JOURNAL_MESSAGE',
+      'peek' => 'MQSEC_PEEK_MESSAGE',
+      'read-permission' => 'MQSEC_GET_QUEUE_PERMISSIONS',
+      'change-permission' => 'MQSEC_CHANGE_QUEUE_PERMISSIONS',
+      'read-property' => 'MQSEC_GET_QUEUE_PROPERTiES',
+      'write-property' => 'MQSEC_SET_QUEUE_PROPERTIES',
+      'read-journal' => 'MQSEC_RECEIVE_JOURNAL_MESSAGE',
+      'read-message' => 'MQSEC_RECEIVE_MESSAGE',
+      'write-message' => 'MQSEC_WRITE_MESSAGE',
+      'delete-message' => 'MQSEC_DELETE_MESSAGE',
+    }.freeze
+
+    def initialize(path)
+      @path = path
+      generate_cache
+    end
+
+    def to_s
+      "Message Queue #{@path.inspect}"
+    end
+
+    def method_missing(method, *)
+      if @cache.respond_to?(:keys) && @cache.keys.include?(method.to_s.downcase)
+        return @cache[method.to_s.downcase]
+      end
+
+      super
+    end
+
+    def respond_to_missing?(method, *)
+      if @cache.respond_to?(:keys)
+        return true if @cache.keys.include?(method.to_s.downcase)
+      end
+
+      super
+    end
+
+    def exists?
+      !@cache.nil?
+    end
+
+    # If negating `#denied?`, use this method instead. A lot may be lost serializing and deserializing the
+    # binary values, so it's best to just have PowerShell handle the comparison.
+    def allowed?(permission, opts = {})
+      script = allowed_wrapper(permission, opts) + "\n\n"
+      script += 'If (($mask -band $target_perm) -eq $target_perm) { Exit 98; } Else { Exit 99; }'
+
+      cmd = inspec.powershell(script)
+
+      case cmd.exit_status
+      when 98 then return true
+      when 99 then return false
+      else
+        # It should never reach here, but for debugging, this will be left in
+        obj = {
+          code: cmd.exit_status,
+          stderr: cmd.stderr,
+          stdout: cmd.stdout,
+        }
+        raise "Problem with P/Invoke: #{obj.inspect}"
+      end
+    end
+
+    # If negating `#allowed?`, use this method instead. A lot may be lost serializing and deserializing the
+    # binary values, so it's best to just have PowerShell handle the comparison.
+    def denied?(permission, opts = {})
+      script = allowed_wrapper(permission, opts) + "\n\n"
```

Comment:
could this be?:
```
!allowed?(permission, opts)
```

Reply:
Possibly...? I'm having trouble thinking of an example where negating the other method's `If (($mask -band $target_perm) -eq $target_perm)` result wouldn't work the same as this.